lib/client/api.go

```diff
@@ -449,19 +460,34 @@ func readProfile(profileDir string, profileName string) (*ProfileStatus, error)
 		clusterName = profile.Name()
 	}
 
+	tlsCert, err := key.TLSCertificate()
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	tlsID, err := tlsca.FromSubject(tlsCert.Subject, time.Time{})
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
 	return &ProfileStatus{
 		ProxyURL: url.URL{
 			Scheme: "https",
 			Host:   profile.WebProxyAddr,
 		},
 		Username:       profile.Username,
-		Logins:         cert.ValidPrincipals,
+		Logins:         sshCert.ValidPrincipals,
 		ValidUntil:     validUntil,
 		Extensions:     extensions,
 		Roles:          roles,
 		Cluster:        clusterName,
 		Traits:         traits,
 		ActiveRequests: activeRequests,
+		// The TLS cert may have k8s users and groups even when proxy isn't
+		// configured to talk to a k8s cluster.
+		// RouteToCluster is only set when k8s support is enabled though.
+		KubeEnabled: tlsID.RouteToCluster != "",
```

Comment:
This is not a reliable indicator that Kubernetes support is enabled. If you are using the `-J` flag for SSH `RouteToCluster` will be set. What about something like the following:

```go
var kubeEnabled bool
if len(tlsID.KubernetesUsers) > 0 || len(tlsID.KubernetesGroups) > 0 {
   kubeEnabled = true
}

Reply:
`KubernetesUsers` is set to user login in the default `admin` role, so that's not a reliable indicator either.
I just realized that https://github.com/gravitational/teleport/pull/4427 will make k8s support explicitly visible in the cert, so I'll wait for that to merge and use the new TLS extension

Reply:
Sounds good to me.